mapboxgl/templates/choropleth_alt.html

```diff
@@ -0,0 +1,204 @@
+{% extends "main.html" %}
+
+<!-- update legend item key style -->
+{% block extra_css %}
+<style type='text/css'>
+    .legend div span { border-radius: 20%; }
+</style>
+{% endblock extra_css %}
+
+{% block javascript %}
+var legend = document.getElementById('legend');
+
+mapboxgl.accessToken = '{{ accessToken }}';
+
+var map = new mapboxgl.Map({
+    container: 'map',
+    style: '{{ styleUrl }}', 
+    center: {{ center }},
+    zoom: {{ zoom }},
+    transformRequest: (url, resourceType)=> {
+        if ( url.slice(0,22) == 'https://api.mapbox.com' ) {
+            //Add Python Plugin identifier for Mapbox API traffic
+            return {
+                url: [url.slice(0, url.indexOf("?")+1), "pluginName=PythonMapboxgl&", url.slice(url.indexOf("?")+1)].join('')
+            }
+        }
+        else {
+            //Do not transform URL for non Mapbox GET requests
+            return {url: url}
+        }
+    }
+});
+
+var joinData = {{ joinData }};
+
+map.addControl(new mapboxgl.NavigationControl());
+
+calcCircleColorLegend({{ colorStops }}, "{{ colorProperty }}");
+
+function calcNumericColorValue(num, arr) {
```

Comment:
Can we remove this helper function if we generate the color interpolation in the Python thread (see comment below)?

Reply:
That's cruft. It's not being used, sorry!